publish.gradle

```diff
@@ -1,132 +1,217 @@
 
-project(':native:wpiutil').build.dependsOn outputVersions
-project(':native:ntcore').build.dependsOn outputVersions
-if (project.buildArm) {
-    project(':arm:wpiutil').build.dependsOn outputVersions
-    project(':arm:ntcore').build.dependsOn outputVersions
-}
+def licenseFile = new File("$rootDir/license.txt")
 
+task cppSourcesZip(type: Zip) {
+    destinationDir = outputsFolder
+    classifier = "sources"
 
+    from(licenseFile) {
+        into '/'
+    }
 
-// We change what repo we publish to depending on whether this is a development, beta, stable, or full
-// release. This is set up in the main gradle file.
-publishing {
-    publications {
-        def nat = project('native:ntcore')
-        if (!project.hasProperty('skipJava')) {
-            java(MavenPublication) {
-                artifact nat.jar
-                artifact nat.networktablesJavaSource
-                artifact nat.networktablesJavadoc
-                if (project.buildArm) {
-                    def natArm = project('arm:ntcore')
-                    artifact natArm.jar
-                    // If the library is not embedded include it in the repo
-                    if (!project.hasProperty('compilerPrefix')) {
-                        artifact natArm.ntcoreZip
-                    }
-                }
+    from('src/main/native/cpp') {
+        into '/'
+    }
 
-                if (project.hasProperty('makeDesktop')) {
-                    artifact nat.jar, {
-                        classifier = 'desktop'
+    model {
+        tasks {
+            it.each {
+                if (it.name.contains('jniHeaders')) {
+                    from (it.outputs.files) {
+                        into '/'
                     }
+                    dependsOn it
                 }
-
-                groupId 'edu.wpi.first.wpilib.networktables.java'
-                artifactId 'NetworkTables'
-                version ntVersion
             }
         }
-        cpp(MavenPublication) {
-            artifact nat.ntcoreZip
-            artifact ntcoreSourceZip
-            if (project.buildArm) {
-                artifact project(':arm:ntcore').ntcoreZip
-            }
+    }
+}
+
+task cppHeadersZip(type: Zip) {
+    destinationDir = outputsFolder
+    classifier = "headers"
+
+    from(licenseFile) {
+        into '/'
+    }
+
+    from('src/main/native/include') {
+        into '/'
+    }
+}
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+artifacts {
+    archives sourcesJar
+    archives javadocJar
+    archives cppHeadersZip
+    archives cppSourcesZip
+}
 
-            if (project.hasProperty('makeDesktop')) {
-                artifact nat.ntcoreZip, {
-                    classifier = 'desktop'
+def createComponentZipTasks = { components, name, base, type, project, func ->
```

Comment:
I'm going to pass on this, because I use a few script globals, and switching to a function breaks that, with the workarounds looking super hacky